executor/adapter.go

```diff
@@ -369,9 +370,13 @@ func (a *ExecStmt) logSlowQuery(txnTS uint64, succ bool) {
 			"[QUERY] %vcost_time:%v %s succ:%v con:%v user:%s txn_start_ts:%v database:%v %v%vsql:%v",
 			internal, costTime, sessVars.StmtCtx.GetExecDetails(), succ, connID, user, txnTS, currentDB, tableIDs, indexIDs, sql)
 	} else {
+		execDetail := sessVars.StmtCtx.GetExecDetails()
 		logutil.SlowQueryLogger.Warnf(
 			"[SLOW_QUERY] %vcost_time:%v %s succ:%v con:%v user:%s txn_start_ts:%v database:%v %v%vsql:%v",
-			internal, costTime, sessVars.StmtCtx.GetExecDetails(), succ, connID, user, txnTS, currentDB, tableIDs, indexIDs, sql)
+			internal, costTime, execDetail, succ, connID, user, txnTS, currentDB, tableIDs, indexIDs, sql)
+		metrics.SlowQueryProcessHistogram.Observe(costTime.Seconds())
+		metrics.SlowQueryCopHistogram.Observe(execDetail.ProcessTime.Seconds())
+		metrics.SlowQueryWaitHistogram.Observe(execDetail.WaitTime.Seconds())
```

Comment:
to eliminate the ambiguity, it's better to:

- s/SlowQueryProcessHistogram/TotalQueryProcHistogram/
- s/SlowQueryWaitHistogram/TotalCopWaitHistogram/
- s/SlowQueryCopHistogram/TotalCopProcHistogram/